### PR TITLE
chore: update german translation

### PR DESCRIPTION
**Pull request by PocketMiner82**
Added missing german locale files + Updated existing ones to allow more words.

### CODE REVIEW

**Comment by JarbasAl** [comment]
hello, apologies for the late reply, can you please resubmit these via https://gitlocalize.com/repo/9609/ instead?

It is our new translation platform

**Comment by JarbasAl** [comment]
still ironing out the quirks of the translation framework, it always requires a reference language to be translated from and currently will have the number of "slots" depending on the english samples

right now handling this is a manual process when we run into issues

if possible try to use the optional syntax `(this|that)` to use less slots, for the reverse you can write `[UNUSED]` to mark a slot as not needing translation (and still count as translated)

i can add a bunch of `[UNUSED]` to the english resources to create more slots too, so if the above doesnt work let me know whenever you run into a file that needs me to extend the slot number

editing the json files should usually not be done, the expectation is that those are handled by the automations and are just a intermediate format to talk with the translation framework, if edited it should be **ONLY en-us** json files. you can add `[UNUSED]` samples to the english .jsons to create new slots, those should be picked up by gitlocalize

**Comment by PocketMiner82** [comment]
Hey @JarbasAl
1. Would it be an option to add like 100 `[UNUSED]` to all english files (not just in ovos-skill-ip) so that we always would have enough slots? using the optional syntax (at least for me) sometimes doesn't really work out. One time I tried to create an intent with only the optional syntax and the training process ended up in an infinite loop...

2. I also edited the code (made digits optional - `.require("last").optionally("digits")`, fixed warning about qml extension). Should I open another PR for this?

**Comment by JarbasAl** [comment]
please split the resource files from the code changes, not sure what impact changing that adapt intent will have so it's good to separate both

i'm not too keen on adding a lot of [UNUSED] around, as every language will need to copy paste those to get 100% translation and that is annoying, i will give this some thinking, perhaps something like 3 [UNUSED] would be acceptable and more would require manual intervention

File: locale/de-de/last.voc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+schließend
+schließende
+schließenden
+schließendes
+schließender
+schließendem
+ende
+enden
+endes
+ender
```
Comment (JarbasAl) [inline locale/de-de/last.voc:31]: would 5 slots suffice here? using the optional syntax i would group these as

(letzt|...)
(hinter|....)
(abschließend|...)
(schließend|...)
(ende|...)

?

**Comment by JarbasAl** [comment]
> One time I tried to create an intent with only the optional syntax and the training process ended up in an infinite loop...

please let us know if this happens, that would be some unknown bug we need to fix!

File: locale/de-de/last.voc (hunk 1 above)
Comment (PocketMiner82) [reply locale/de-de/last.voc:31]: yes that works, i tested it. btw it also works without the parentheses

Comment (PocketMiner82) [reply locale/de-de/last.voc:31]: It also works in just one line, so you maybe don't have to add more lines to the english file

**Comment by PocketMiner82** [comment]
> > One time I tried to create an intent with only the optional syntax and the training process ended up in an infinite loop...
> 
> please let us know if this happens, that would be some unknown bug we need to fix!

@JarbasAl I managed to reproduce it:

First intent:
```
(schalten|schaltet|schalte|schalter|schalt|machen|mache|macht|macher|mach|stellen|stelle|stell|stellt|steller|) (bitte|) (den|die|das|) {entity} (bitte|) (ein|an|einschalten|einschaltet|einschalte|einschalter|einmach|einschalt|einmachen|einmache|einmacht|einmacher|einstellen|einstelle|einstell|einstellt|einsteller|anschalten|anschaltet|anschalte|anschalter|anmach|anschalt|anmachen|anmache|anmacht|anmacher|anstellen|anstelle|anstell|anstellt|ansteller) (schalten|schaltet|schalte|schalter|schalt|machen|mache|macht|macher|mach|stellen|stelle|stell|stellt|steller|) (bitte|)
```

Second intent:
```
{entity} (bitte|) (ein|an|einschalten|einschaltet|einschalte|einschalter|einmach|einschalt|einmachen|einmache|einmacht|einmacher|einstellen|einstelle|einstell|einstellt|einsteller|anschalten|anschaltet|anschalte|anschalter|anmach|anschalt|anmachen|anmache|anmacht|anmacher|anstellen|anstelle|anstell|anstellt|ansteller) (bitte|)
{entity} (bitte|) (ein|an) (schalten|schaltet|schalte|schalter|schalt|machen|mache|macht|macher|mach|stellen|stelle|stell|stellt|steller|) (bitte|)
(schalten|schaltet|schalte|schalter|schalt|machen|mache|macht|macher|mach|stellen|stelle|stell|stellt|steller) (bitte|) (den|die|das|) {entity} (bitte|) (an|ein) (bitte|)
```

From https://github.com/PocketMiner82/skill-homeassistant/blob/20.08/vocab/de-de/turn.on.intent

The issue is that one CPU core is always on 100% when I use the first intent. And it takes ages for OVOS to respond. When I use the second intent, the CPU usage looks normal and it responds immediately. When I had this issue a few months ago, it wouldn't even respond at all.

File: locale/de-de/last.voc (hunk 1 above)
Comment (PocketMiner82) [reply locale/de-de/last.voc:31]: I will try to put them in one line wherever possible, will tell you when it is not

**Comment by PocketMiner82** [comment]
I think you can close this PR if you want, will update the translations on GitLocalize soon.

**Comment by JarbasAl** [comment]
> @JarbasAl I just figured out that it is possible to add multiple lines in one translation entry. would that work?

you know what, thats a good test, let me see what that would look like in the UI side, the automations should create valid files and handle the `\n` characters indicating a new line. 

I will want to tweak the automations regardless to have a nice .json file if we adopt this, but let's do a test and see if it works as intended!

**Comment by PocketMiner82** [comment]
@JarbasAl I updated all translations on GitLocalize.

**Pull request closed without merging**